crates/oxc_linter/src/rules/typescript/no_extraneous_class.rs

```diff
@@ -320,5 +328,17 @@ fn test() {
         ("abstract class Foo { constructor() {} }", None),
     ];
 
-    Tester::new(NoExtraneousClass::NAME, NoExtraneousClass::PLUGIN, pass, fail).test_and_snapshot();
+    let fix = vec![
+        ("class Foo {}", "", None, FixKind::Suggestion),
```

Comment:
Can you please add a test for `export class Foo {}` too?
I guess this will result into `export` and will lead into invalid syntax.
Deleting the export will not be safe because other files could use the import.

Maybe we should mark them as dangerous suggestions when they have an export parent?

Reply:
I have made it always a dangerous suggestion, and fixed the `export class` case

If a user had code that referenced this class, removeing the class (even if there's no export) is dangrous, as the IdentiferRefernece now refernces a symbol that no longer exists.